altsrc/map_input_source.go

```diff
@@ -149,19 +177,55 @@ func (fsm *MapInputSource) Int64(name string) (int64, error) {
 	return 0, nil
 }
 
+func castToUint(v interface{}) (uint, bool) {
+	uintValue := uint(0)
+	isType := false
+	if v == nil {
+		return uintValue, true
+	}
+
+	// There are only four cases(int, int64, uint64, float64) when parsing the integer in yaml.v3 pkg
+	// But the last case, float64, is an error case so that ignored
+	vType := reflect.TypeOf(v).Kind()
+	switch vType {
+	case reflect.Int:
+		intValue := v.(int)
+		if intValue >= 0 {
+			uintValue = uint(intValue)
+			isType = true
+		}
+	case reflect.Int64:
+		int64Value := v.(int64)
+		if int64Value >= 0 && uint64(int64Value) <= math.MaxUint {
```

Comment:
Add/update test case to test this condition

Reply:
It needs the 32bits platform to cover this.